zeppelin-zengine/src/main/java/org/apache/zeppelin/scheduler/RemoteScheduler.java

```diff
@@ -82,10 +88,11 @@ public void runJobInScheduler(Job job) {
    * RUNNING status. This thread will exist after job is in RUNNING/FINISHED state.
    */
   private class JobStatusPoller extends Thread {
```

Comment:
In the following, the interrupt behaviour of the StatusPoller has been improved.